changelog/dpath.dd

```diff
@@ -0,0 +1,41 @@
+DUB settings & packages directory placement overhauled
+
+You can now configure where DUB places its downloaded packages and where the user configuration is stored through environment variables or through the dub configuration. You need to use an environment variable or the system-wide dub configuration to specify where the user configuration is stored.
+
+By default DUB stores the packages on
+- Windows: `%APPDATA%/dub/{packages/,settings.json}`
```

Comment:
that is for localRepository, but I think the more used thing is the userRepository. I don't think localRepository is ever used unless the user explicitly wants to store things in the local repository.

For simplification in the config, the config overrides both the user and the local repository.

Reply:
`dub fetch` (and related `fetch` commands) use `localRepository`:
https://github.com/dlang/dub/blob/17d8c9467b81cf2b2ac57c338f01f71c2c71b013/source/dub/dub.d#L768-L773

The logic behind it, as I very recently understood and documented, is that using APPDATA for settings is good (you want your settings to roam), but not for packages (because it might use a lot of space / lead to complications with stale binaries):
https://github.com/dlang/dub/blob/17d8c9467b81cf2b2ac57c338f01f71c2c71b013/source/dub/dub.d#L1748-L1750

Reply:
IOW:
```diff
-- Windows: `%APPDATA%/dub/{packages/,settings.json}`
+- Windows: `%APPDATA%/dub/settings.json` 
+- Windows: `%LOCALAPPDATA%/dub/packages/` 
```

I don't know if that should affect the functionality too.

Reply:
oh I see now, I thought localSettings correspond to PlacementLocation.local and userSettings to PlacementLocation.user, but it turns out to be a bit different there, which is confusing. Will update the changelog then.